modules/tests/test_ref_modules/test_ref_modules.cpp

```diff
@@ -575,5 +575,493 @@ TEST_F(RefModulesTest, UpdateDevicePower)
+        auto expectedClassifierDelata = static_cast<Float>(blockSize) / static_cast<Float>(deviceResolution);
+
+        ASSERT_EQ(classifierDelta, expectedClassifierDelata);
+    }
+    else
+    {
+        ASSERT_EQ(classifierDomainRule.getType(), DataRuleType::Explicit);
+    }
+}
+
+TEST_F(RefModulesTest, ClassifierCheckBuildinClass)
+{
+    const auto instance = Instance();
+
+    const auto device = instance.addDevice("daqref://device0");
+
+    auto customClassList = List<Float>(1, 10, 100, 1000);
+    auto blockSize = 2;
+    auto classCount = 2;
+    auto inputRange = Range(-20.0, 20.0);
+
+    const auto classifierFb = instance.addFunctionBlock("ref_fb_module_classifier");
+    classifierFb.setPropertyValue("UseCustomClasses", false);
+    classifierFb.setPropertyValue("CustomClassList", customClassList);
+    classifierFb.setPropertyValue("BlockSize", blockSize);
+    classifierFb.setPropertyValue("ClassCount", classCount);
+
+    const auto deviceChannel = device.getChannels()[0];
+    deviceChannel.setPropertyValue("CustomRange", inputRange);
+    const auto deviceSignal = deviceChannel.getSignals()[0];
+
+    const auto classifierInputPort = classifierFb.getInputPorts()[0];
+    classifierInputPort.connect(deviceSignal);
+
+    const auto classifierSignal = classifierFb.getSignals()[0];
+    const auto classifierSignalDescription = classifierSignal.getDescriptor();
+
+    // Classifier returning values with float float
+    ASSERT_EQ(classifierSignalDescription.getSampleType(), SampleType::Float64);
+
+    // Dimension of classifier's output signal have to be 1 (vector)
+    ASSERT_EQ(classifierSignalDescription.getDimensions().getCount(), 1);
+
+    // Classifier returns values in range [0, 1]
+    ASSERT_EQ(classifierSignalDescription.getValueRange(), Range(0, 1));
+
+    // Classifier returns values are %
+    ASSERT_EQ(classifierSignalDescription.getUnit().getSymbol(), "%");
+
+    // Checking dimension of classifier
+    {
+        auto signalDimension = classifierSignalDescription.getDimensions()[0];
+        auto labels = signalDimension.getLabels();
+        Float expectedMark = inputRange.getLowValue();
+        for (const auto& mark : labels)
+        {
+            ASSERT_EQ(static_cast<Float>(mark), expectedMark);
+            expectedMark += classCount;
+        }
+        ASSERT_EQ(labels[labels.getCount() - 1], inputRange.getHighValue());
+    }
+
+    const auto classifierDomainSignal = classifierSignal.getDomainSignal();
+    const auto classifierDomainSignalDescription = classifierDomainSignal.getDescriptor();
+
+    const auto deviceDomainDescriptor = deviceSignal.getDomainSignal().getDescriptor();
+
+    const auto deviceDomainRule = deviceDomainDescriptor.getRule();
+    const auto classifierDomainRule = classifierDomainSignalDescription.getRule();
+
+    // Check Linear Rule
+    if (deviceDomainRule.getType() == DataRuleType::Linear)
```

Comment:
Is this `if` statement needed? (similar question for the checks in the other tests)